### ISSUE TITLE
Shell history not working in cmd.exe or Python REPL

### ISSUE THREAD

**Issue by nikshankarnoble** [nerdvegas/rez#1234]
Hi,

With Rez 2.103.4 (Python 3.7) on Windows 10, we are finding that using cmd.exe with Rez doesn't let you step through your command history with the up-arrow key. This issue was reported a couple of years ago but seemed to have been fixed in 2.35.0
https://github.com/nerdvegas/rez/issues/616

```
$ rez env
> $ echo Hello World!
> $ # hit up-arrow
> $ # no response.. :(
```

It works as expected in Powershell until you launch a Python REPL process, which it then no longer works inside of.

Any suggestions/help would be very much appreciated! :) Apologies I don't really have the knowledge to look into this myself.

Thanks,
Nikhil

**Comment by nerdvegas** [comment]
I'm going to regard this as a feature request given current lack of support for managing shell history. Please see https://github.com/nerdvegas/rez/issues/1240

**Comment by instinct-vfx** [comment]
While i agree with the notion that this should be managed actively there is a specific issue with cmd that breaks this. It works with Terminal and Powershell as expected. I THINK Marcus did a fix ages ago and it was choosing different flags when calling cmd. But that might of course have other side effects. I can take a look if i find the time.

**Comment by nerdvegas** [comment]
Yeah I looked back at that old PR, and it seems like fixing cmd history was
(for whatever reason) a side-effect of removing that interstitial cmd proc.
Not sure what may have changed in the meantime.

I'll reopen this issue but for anyone working on it please refer to #1240
in the fixing branch name, cheers!

Thx
A

**Comment by nerdvegas** [comment]
Please note that this should be fixed with general solution in mind - https://github.com/nerdvegas/rez/issues/1240

**Comment by instinct-vfx** [comment]
As your link indicates i would argue this is more of a console host and shell issue than necessarily a rez issue. I always forget this was an issue because i use Windows Terminal as by console host and a mix of cmd and powershell and history is just fine even in nested shells.

**Comment by herronelou** [comment]
I don't disagree, it's just not particularly obvious at first, where launching commands from cmd or powershell has a properly working history, but as soon as rez is in the mix it seemingly suddenly stops working. That made me believe that Rez was at fault, and if I'm not mistaken, that's also what others might be believing based on this issue and #616 #1240 and #1320 

It could also explain that removing a layer of subprocess via #627 has seemingly made it work a bit better, as that reduced the buffer stack by one, but that going any further down would still break.

If somebody else can confirm that this is indeed the issue that was mentioned, even though it's not a Rez issue, it might be worth mentioning somewhere in the docs how to workaround?

**Comment by instinct-vfx** [comment]
Absolutely should be mentioned in the docs. Shell plugin specific documentation and pitfalls would generally be a good addition to the docs as even just the differences between cmd and pwsh/powershell can be confusing.

**Comment by instinct-vfx** [comment]
Would you mind adding it to the documentation improvement discussion thread?

**Comment by jtomori** [comment]
Hello,
I came across the same issue with Rez 2.111.3 on Windows 10 and cmd.exe.
Setting _Number of Buffers_ to 12 in the cmd properties helps with commands history in a rez context and also in Python REPL. 

Attaching screenshot of cmd properties, so that it's easier to find fix in this thread.

<img src="https://user-images.githubusercontent.com/22801191/204554360-539426d7-63df-4b5a-b2af-e7f1b2e32175.png" width="350px">

Interestingly commands history worked well in IPython even with the default _Number of Buffers_ setting of 4.
